File: src/extension/compass/__tests__/fixtures/type_hints.cs
namespace Demo
{
    public class OrganizationService
    {
    }

    public class Logger
    {
    }

    public class Repository
    {
    }

    public class TenantController
    {
        private Logger _logger;
        public Repository Repo { get; set; }

        public TenantController(OrganizationService service)
        {
        }

        public OrganizationService Resolve(Repository repo)
        {
            return null;
        }

        public string Count(int value)
        {
            return value.ToString();
        }

        public System.Threading.Tasks.Task<Repository> LoadAsync()
        {
            return null;
        }
    }
}

File: src/extension/compass/__tests__/fixtures/type_hints.kt
class OrganizationService

class Logger

class Repository

class TenantController(private val service: OrganizationService) {
    private val logger: Logger = Logger()

    fun resolve(repo: Repository): OrganizationService {
        return service
    }

    fun count(value: Int): String {
        return value.toString()
    }
}
